src/ppl/src/IOPlacer.cpp

```diff
@@ -684,14 +683,14 @@ int IOPlacer::micronsToDbu(double microns)
   return (int64_t) (microns * getBlock()->getDbUnitsPerMicron());
 }
 
-void IOPlacer::writePinPlacement()
+void IOPlacer::writePinPlacement(const char* file_name)
 {
-  std::string file_name = parms_->getPinPlacementFile();
-  if (file_name.empty()) {
+  std::string filename = file_name;
+  if (filename.empty()) {
     return;
   }
 
-  std::ofstream out(file_name);
+  std::ofstream out(filename);
```

Comment:
Does an error get thrown if the file can't be opened?

Reply:
It isn't related to this PR but it is a good point.  We should check 'if (!out)'  ...